sklearn/cluster/k_medoids_.py

```diff
@@ -0,0 +1,284 @@
+
+            if sum(cluster_ics == cluster_idx) == 0:
+                warnings.warn("Cluster {} is empty!".format(cluster_idx))
+                continue
+
+            # Find current cost that is associated with cluster_idx.
+            # Cost is the sum of the distance from the cluster
+            # members to the medoid.
+            curr_cost = np.sum(D[medoid_ics[cluster_idx],
+                                 cluster_ics == cluster_idx])
+
+            # Extract the distance matrix between the data points
+            # inside the cluster_idx
+            D_in = D[cluster_ics == cluster_idx, :]
+            D_in = D_in[:, cluster_ics == cluster_idx]
+
+            # Calculate all costs there exists between all
+            # the data points in the cluster_idx
+            all_costs = np.sum(D_in, axis=1)
+
+            # Find the index for the smallest cost in cluster_idx
+            min_cost_idx = np.argmin(all_costs)
+
+            # find the value of the minimum cost in cluster_idx
+            min_cost = all_costs[min_cost_idx]
+
+            # If the minimum cost is smaller than that
+            # exhibited by the currently used medoid,
+            # we switch to using the new medoid in cluster_idx
+            if min_cost < curr_cost:
+
+                # Find data points that belong to cluster_idx,
+                # and assign the newly found medoid as the medoid
+                # for cluster c
+                medoid_ics[cluster_idx] = \
+                    np.where(cluster_ics == cluster_idx)[0][min_cost_idx]
+
+    def transform(self, X):
+        """Transforms X to cluster-distance space.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Data to transform.
+
+        Returns
+        -------
+        X_new : array, shape=(n_samples, n_clusters)
+            X transformed in the new space.
+        """
+
+        self._check_is_fitted()
+
+        # Apply distance metric wrt. cluster centers (medoids),
+        # and return these distances
+        return self.distance_func(X, Y=self.cluster_centers_)
+
+    def predict(self, X):
+
+        self._check_is_fitted()
+
+        # Check that the array is good and attempt to convert it to
+        # Numpy array if possible
+        X = check_array(X)
+
+        # Apply distance metric wrt. cluster centers (medoids)
+        D = self.distance_func(X, Y=self.cluster_centers_)
+
+        # Assign data points to clusters based on
+        # which cluster assignment yields
+        # the smallest distance
+        labels = np.argmin(D, axis=1)
+
+        return labels
+
+    def inertia(self, X):
+
+        # Map the original X to the distance-space
+        Xt = self.transform(X)
+
+        # Define inertia as the sum of the sample-distances
+        # to closest cluster centers
+        inertia = np.sum(np.min(Xt, axis=1))
+
+        return inertia
+
+    def _check_is_fitted(self):
```

Comment:
I meant that ``_check_is_fitted`` is a one-line method, so I'd rather not define a new method for it.